core/types.ml

```diff
@@ -136,6 +136,7 @@ type lens_phrase =
   | In        of Lens_operators.name list * (Constant.constant list) list
   | Case      of lens_phrase option * (lens_phrase * lens_phrase) list * lens_phrase
   | TupleLit  of lens_phrase list
+and lens_phrase = lens_phrasenode SourceCode.With_pos.t
```

Comment:
I am not a fan of this change. Now our type algebra depends on surface syntax matters.

Reply:
I agree that something in types.ml knowing about source code positions seems unclean in terms of separating the different stages of compilation